WORKSPACE

```diff
@@ -42,19 +42,43 @@ Try running `yarn bazel` instead.
 )
 
 #
-# Load and install our dependencies downloaded above.
+# Since we don't have @npm//@bazel/foobar npm packages to setup
+# the npm_bazel_foobar repositories, we set these up manually as
+# local repositories.
 #
 
 local_repository(
-    name = "examples_program",
-    path = "examples/program",
+    name = "npm_bazel_jasmine",
+    path = "packages/jasmine/src",
 )
 
 local_repository(
-    name = "internal_e2e_packages",
-    path = "internal/e2e/packages",
+    name = "npm_bazel_karma",
+    path = "packages/karma/src",
 )
 
+local_repository(
+    name = "npm_bazel_labs",
+    path = "packages/labs/src",
+)
+
+local_repository(
+    name = "npm_bazel_typescript",
+    path = "packages/typescript/src",
```

Comment:
I'm still a bit confused, it seems like we still have nested workspaces, so what's the magic that lets us simplify the scripts and CI? just that there's a directory above the workspace so the packages can refer to each other using build_bazel_rules_nodejs labels, then that top dir uses the nested workspace labels?

Reply:
main reason is to support the `http_archive` use can for users that don't want to use `npm_bazel_foobar` without `@bazel/foobar`. 

```
http_archive(name = "npm_bazel_foobar", url="...", ...)
```

The tests and docs can be outside of that workspacer but the rules themselves and their dependencies on other `npm_bazel_other` rules need to be in a `npm_bazel_foobar` namespace for this usecase.

Reply:
Also, all of the test targets for the packages are the root workspace `build_bazel_rules_nodejs` (under `/packages/foobar/test` by convetion) so `bazel test ...` will test everything and also run all of the build targets in the nested `npm_bazel_other` workspaces as those are deps of the test targets.

The nested workspaces themselves are under `/packages/foobar/src` by convention and there is a `/packages/foobar/src/defaults.bzl` which provides macros for the rules provided to use in the downstream http_archive or local development use case. Could probably use a better name than `defaults.bzl`?

All local tests and packages that depend on other packages use the `defaults.bzl` overrides while e2e tests & examples and downstream consumers of the npm packages will use the `index.bzl` rules.